exchanges/huobi/huobi_wrapper.go

```diff
@@ -287,81 +329,190 @@ func (h *HUOBI) Run() {
-}
 
-// UpdateTicker updates and returns the ticker for a currency pair
-func (h *HUOBI) UpdateTicker(p currency.Pair, assetType asset.Item) (*ticker.Price, error) {
-	tickers, err := h.GetTickers()
+	coinmarginedFuturesPairs, err := h.FetchTradablePairs(asset.CoinMarginedFutures)
 	if err != nil {
-		return nil, err
+		return err
 	}
-	pairs, err := h.GetEnabledPairs(assetType)
+	cp, err := currency.NewPairsFromStrings(coinmarginedFuturesPairs)
 	if err != nil {
-		return nil, err
+		return err
 	}
-	for i := range pairs {
-		for j := range tickers.Data {
-			pairFmt, err := h.FormatExchangeCurrency(pairs[i], assetType)
-			if err != nil {
-				return nil, err
-			}
+	return h.UpdatePairs(cp, asset.CoinMarginedFutures, false, forceUpdate)
+}
 
-			if pairFmt.String() != tickers.Data[j].Symbol {
-				continue
+// UpdateTicker updates and returns the ticker for a currency pair
+func (h *HUOBI) UpdateTicker(p currency.Pair, assetType asset.Item) (*ticker.Price, error) {
+	if !h.SupportsAsset(assetType) {
+		return nil, fmt.Errorf("asset type of %s is not supported by %s", assetType, h.Name)
+	}
+	switch assetType {
+	case asset.Spot:
+		tickers, err := h.GetTickers()
+		if err != nil {
+			return nil, err
+		}
+		pairs, err := h.GetEnabledPairs(assetType)
+		if err != nil {
+			return nil, err
+		}
+		for i := range pairs {
+			for j := range tickers.Data {
+				pairFmt, err := h.FormatExchangeCurrency(pairs[i], assetType)
+				if err != nil {
+					return nil, err
+				}
+				if !strings.EqualFold(pairFmt.Lower().String(), tickers.Data[j].Symbol) {
+					continue
+				}
+				err = ticker.ProcessTicker(&ticker.Price{
+					High:         tickers.Data[j].High,
+					Low:          tickers.Data[j].Low,
+					Volume:       tickers.Data[j].Volume,
+					Open:         tickers.Data[j].Open,
+					Close:        tickers.Data[j].Close,
+					Pair:         pairs[i],
+					ExchangeName: h.Name,
+					AssetType:    assetType})
+				if err != nil {
+					return nil, err
+				}
 			}
+		}
+	case asset.CoinMarginedFutures:
+		fmtPair, err := h.FormatExchangeCurrency(p, assetType)
+		if err != nil {
+			return nil, err
+		}
+		marketData, err := h.GetSwapMarketOverview(fmtPair.String())
+		if err != nil {
+			return nil, err
+		}
 
-			err = ticker.ProcessTicker(&ticker.Price{
-				High:         tickers.Data[j].High,
-				Low:          tickers.Data[j].Low,
-				Volume:       tickers.Data[j].Volume,
-				Open:         tickers.Data[j].Open,
-				Close:        tickers.Data[j].Close,
-				Pair:         pairs[i],
-				ExchangeName: h.Name,
-				AssetType:    assetType})
-			if err != nil {
-				return nil, err
-			}
+		err = ticker.ProcessTicker(&ticker.Price{
+			High:         marketData.Tick.High,
+			Low:          marketData.Tick.Low,
+			Volume:       marketData.Tick.Vol,
+			Open:         marketData.Tick.Open,
+			Close:        marketData.Tick.Close,
+			Pair:         p,
+			Bid:          marketData.Tick.Bid[0],
+			Ask:          marketData.Tick.Ask[0],
```

Comment:
You have not checked whether Bid or Ask even has any entries. This could lead to a panic